components/PlayerForm.js

```diff
@@ -46,8 +54,76 @@ export default function PlayerForm({ onSubmit, formName, defaultData }) {
     fetchAddressInfos(name, value);
   }
 
+  // Cloudinary Upload handler
+
+  function handleFileChange(event) {
+    const file = event.target.files[0];
+    setImage(file);
+    setImageValue(event.target.value);
+  }
+
+  async function handleFileUpload(event) {
+    event.preventDefault();
+    setIsUploading(true);
+
+    const formData = new FormData();
+
+    formData.append('file', image);
+    formData.append('upload_preset', process.env.NEXT_PUBLIC_UPLOAD_PRESET);
+    try {
+      const response = await fetch(
+        `https://api.cloudinary.com/v1_1/${process.env.NEXT_PUBLIC_CLOUDNAME}/upload`,
+        {
+          method: 'POST',
+          body: formData,
+        }
+      );
+      const json = await response.json();
+      const secureUrl = json.secure_url;
+
+      setUploadedImage(json);
+
+      setImageUrl(secureUrl)
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setIsUploading(false);
+      setImage(null);
+      setImageValue('');
+    }
+  }
+
   return (
     <FormContainer aria-labelledby={formName} onSubmit={handleSubmit}>
+      <label htmlFor="playerImage">Player Image:</label>
+      <StyledImageWrapper>
+        <StyledImagePrev src={imageUrl ? imageUrl : defaultData?.playerImage} width={150} height={150} alt="" />
+        {image && (
+          <Image
+            src={URL.createObjectURL(image)}
+            width={150}
+            height={150}
+            alt="Preview Avatar"
+            style={{ objectFit: 'cover' }}
+          />
+        )}
+      </StyledImageWrapper>
+      <StyledInputUpload
+        type="file"
+        id="avatar"
+        onChange={handleFileChange}
+        value={imageValue}
+      />
+      <StyledButton variant={'primary'} width={300} type="button" onClick={handleFileUpload} disabled={!image}>
```

Comment:
At the moment i dont see any visual change when the button is `disabled`. It might be worth adding that to the `StyledButton`

Reply:
thanks for hint